src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php

```diff
@@ -439,6 +439,28 @@ public function testDumpedCsvEnvParameters()
         $this->assertSame(array('foo', 'bar'), $container->getParameter('hello'));
     }
 
+    public function testDumpedCatchEnvParameters()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('fallback_param', 'baz');
+        $container->setParameter('fallback_env', '%env(foobar)%');
+        $container->setParameter('env(foobar)', 'foobaz');
+        $container->setParameter('env(foo)', '{"foo": "bar"}');
+        $container->setParameter('hello', '%env(catch:bar:fallback_param)%');
+        $container->setParameter('hello-bar', '%env(catch:key:baz:json:foo:fallback_env)%');
```

Comment:
i.e. can we add a test to show what will happen if it's `catch:key:baz:json:foo`

Reply:
same comment as https://github.com/symfony/symfony/pull/28976#discussion_r230687605 (refactored)